File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int l = 0;
        int r = 0;
        int maxLen = 0;
        vector<int>hashMap(256, -1);
        while(r < s.size()){
            if(hashMap[s[r]] != -1 && l <= hashMap[s[r]]){
                l = hashMap[s[r]] + 1;
            }
            int len = r - l + 1;
            maxLen = max(maxLen, len);
            hashMap[s[r]] = r;
            r++;
        }
        return maxLen;
    }
};

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int l = 0, r = 0;
        int[] arr =  new int[256];
        Arrays.fill(arr, -1);
        int maxLen = 0;
        while(r < s.length()){
            char ch = s.charAt(r);
            if(arr[ch] != -1 && l <= arr[ch]){
                l = arr[ch] + 1;
            }
            int len = r - l + 1;
            maxLen = Math.max(maxLen, len);
            arr[ch] = r;
            r++;
        }
        return maxLen;  
    }
}